### PR TITLE
Implement DataFrame.dot supporting Series parameter

### PR DESCRIPTION
**Pull request by xinrong-databricks**
Compute the matrix multiplication between the DataFrame and other **series only**.

```
        >>> kdf = ks.DataFrame([[0, 1, -2, -1], [1, 1, 1, 1]])
        >>> kser = ks.Series([1, 1, 2, 1])
        >>> kdf.dot(kser)
        0   -4
        1    5
        dtype: int64

        Note how shuffling of the objects does not change the result.

        >>> kser2 = kser.reindex([1, 0, 2, 3])
        >>> kdf.dot(kser2)
        0   -4
        1    5
        dtype: int64
```

### CODE REVIEW

File: databricks/koalas/frame.py [hunk 1]
```diff
@@ -4008,6 +4008,65 @@ def duplicated(self, subset=None, keep="first") -> "Series":
...
+        >>> kser2 = kser.reindex([1, 0, 2, 3])
+        >>> kdf.dot(kser2)
+        0   -4
+        1    5
+        dtype: int64
+
+        >>> reset_option("compute.ops_on_diff_frames")
+        """
+        if not isinstance(other, ks.Series):
+            raise TypeError("Unsupported type {}".format(type(other).__name__))
```
Comment (itholic) [inline databricks/koalas/frame.py:4066]: Seems pandas also support many other types as below, not only `Series`.

```python
>>> pdf
   0  1  2  3
0  0  1 -2 -1
1  1  1  1  1
>>> pser
0    1
1    1
2    2
3    1
dtype: int64
>>>
>>> pdf.dot(pser.to_frame())  # DataFrame
   0
0 -4
1  5
>>> pdf.dot(pser.index)  # Index
0   -6
1    6
dtype: int64
>>> pdf.dot([1, 2, 3, 4])  # list
0    -8
1    10
dtype: int64
```

Comment (xinrong-databricks) [reply databricks/koalas/frame.py:4066]: Yeah, I should've clarified in the PR description. This PR targets to support other as `Series` only. 

Considering the implementation complexity and user demand, we might want to support other data types later.

Let me modify the PR description and leave a comment for this.

Thanks for looking into this!

Comment (itholic) [reply databricks/koalas/frame.py:4067]: Oh, sure I got it :)

File: databricks/koalas/frame.py [hunk 2]
```diff
@@ -4008,6 +4008,66 @@ def duplicated(self, subset=None, keep="first") -> "Series":
...
 
+    # TODO: support other as DataFrame or array-like
+    def dot(self, other: "Series") -> "Series":
+        """
+        Compute the matrix multiplication between the DataFrame and other.
+
+        This method computes the matrix product between the DataFrame and the
+        values of an other Series
+
+        It can also be called using ``self @ other`` in Python >= 3.5.
```
Comment (ueshin) [inline databricks/koalas/frame.py:4019]: We need to define `__matmul__` for this as same as `Series`?

https://github.com/databricks/koalas/blob/026262dcd7e455be135b7c1e0971768c3c9d37db/databricks/koalas/series.py#L4820-L4824

File: databricks/koalas/frame.py [hunk 3]
```diff
@@ -4008,6 +4008,66 @@ def duplicated(self, subset=None, keep="first") -> "Series":
...
+        Examples
+        --------
+        >>> from databricks.koalas.config import set_option, reset_option
+        >>> set_option("compute.ops_on_diff_frames", True)
+        >>> kdf = ks.DataFrame([[0, 1, -2, -1], [1, 1, 1, 1]])
+        >>> kser = ks.Series([1, 1, 2, 1])
+        >>> kdf.dot(kser)
+        0   -4
+        1    5
+        dtype: int64
```
Comment (ueshin) [inline databricks/koalas/frame.py:4054]: Shall we add an example to use `@` operator?

File: databricks/koalas/frame.py [hunk 4]
```diff
@@ -4008,6 +4008,66 @@ def duplicated(self, subset=None, keep="first") -> "Series":
...
+    # TODO: support other as DataFrame or array-like
+    def dot(self, other: "Series") -> "Series":
+        """
+        Compute the matrix multiplication between the DataFrame and other.
+
+        This method computes the matrix product between the DataFrame and the
+        values of an other Series
+
+        It can also be called using ``self @ other`` in Python >= 3.5.
+
```
Comment (ueshin) [inline databricks/koalas/frame.py:4020]: Shall we add the same note as `transpose` as this uses it internally?

File: databricks/koalas/frame.py (hunk 2 above)
Comment (xinrong-databricks) [reply databricks/koalas/frame.py:4019]: Sounds good!

File: databricks/koalas/frame.py (hunk 3 above)
Comment (xinrong-databricks) [reply databricks/koalas/frame.py:4054]: Added!

File: databricks/koalas/frame.py (hunk 4 above)
Comment (xinrong-databricks) [reply databricks/koalas/frame.py:4020]: Makes sense. Added it!

**Review by ueshin — APPROVED** [review]
LGTM.

**Comment by ueshin** [comment]
Thanks! merging.

**Pull request merged**